src/main/java/com/autotune/analyzer/experiment/ExperimentInitiator.java

```diff
@@ -92,5 +95,35 @@ public void validateAndUpdateResults(
             validationOutputData.setMessage("Exception occurred while validating the result data: " + e.getMessage());
             validationOutputData.setErrorCode(HttpServletResponse.SC_INTERNAL_SERVER_ERROR);
         }
+        return validationOutputData;
+    }
+
+    // Generate recommendations and add it to the kruize object
+    public boolean generateAndAddRecommendations(Map<String, KruizeObject> experimentsMap, List<String> experimentNameList) {
+        if (null == experimentNameList)
+            return false;
+        if (experimentNameList.size() == 0)
+            return false;
+        for (String experimentName: experimentNameList) {
+            if (!experimentsMap.containsKey(experimentName))
+                return false;
+        }
+        for (String experimentName : experimentNameList) {
+            KruizeObject kruizeObject = experimentsMap.get(experimentName);
+            if (AnalyzerConstants.PerformanceProfileConstants.perfProfileInstances.containsKey(kruizeObject.getPerformanceProfile())) {
+                try {
+                    PerfProfileInterface perfProfileInstance =
+                            (PerfProfileInterface) AnalyzerConstants.PerformanceProfileConstants
+                                    .perfProfileInstances.get(kruizeObject.getPerformanceProfile())
+                                    .getDeclaredConstructor().newInstance();
+                    perfProfileInstance.recommend(kruizeObject);
+                } catch (InstantiationException | IllegalAccessException | InvocationTargetException | NoSuchMethodException e ) {
+                    e.printStackTrace();
+                }
+            }
```

Comment:
What happens in the else case here?

Reply:
Will be caught in the validation of performance profile, just added a safety check here.